### DISCUSSION TITLE
Add empty page detection and removal

### DISCUSSION THREAD

**Discussion by lenucksi** [jonaswinkler/paperless-ng#690] [Feature Requests]
##  Short 

Right now, Paperless-ng can do rotation and deskewing of pages. I would like to also be able to automatically have empty pages removed.

## Longer

I use an ADF scanner, often in duplex mode. More often than not pages (A4 sheets of paper) are not consistently printed on both sides. Scanning this with a duplex ADF leads to blank images in the resulting file.
I would like to have those resulting images dropped from the PDF.

**Comment by jonaswinkler** [comment]
Rotation and deskewing are provided by `OCRmyPDF`. I've searched their documentation, and they don't have an option for blank page removal.

Are there any python-compatible tools out there that are able to do this?

**Reply by lenucksi** [reply]
I actually found a GitHub issue around this point for OCRmyPDF: https://github.com/jbarlow83/OCRmyPDF/issues/12 

Also, this StackOverflow hints to PIL as library with extra-sauce: https://dsp.stackexchange.com/questions/48826/fast-empty-page-detection
This piece of Python SW does appear to have blank page extraction, maybe something can be extracted from it: https://pypi.org/project/scanpdf/ (Looks like Apache Licensed)

Maybe some of this can help?

**Reply by lumnikemel** [reply]
This turns out to be somewhat hard, and maybe not something to rely on. My work around was to add a 1-sided and 2-sided option on my scanner profiles to select which option I want based on whether the document is 1-sided or 2-sided. Just my 2 cents.

**Comment by thomassaffer** [comment]
From my point of view manual deletion of empty pages could be a good start ;-) Would be great!
